plugin/src/test/groovy/org/gradle/testretry/AbstractPluginFuncTest.groovy

```diff
@@ -78,6 +78,24 @@ abstract class AbstractPluginFuncTest extends Specification {
                         throw new java.io.UncheckedIOException(e);
                     }
                 }
+
+                public static void flakyAssertPassFailPass(String id) {
+                    Path marker = Paths.get("build/marker.file." + id);
+                    try {
+                        if (Files.exists(marker)) {
+                            int counter = Integer.parseInt(new String(Files.readAllBytes(marker)));
```

Comment:
You probably need the `groovy-nio` as test dependency for the nice `Path` extension methods.
```suggestion
                            int counter = Integer.parseInt(marker.text));
```

Reply:
Same here, it is a plain Java class written alongside test classes.